src/wire/icmpv4.rs

```diff
@@ -441,7 +442,9 @@ impl<'a> Repr<'a> {
 
     /// Emit a high-level representation into an Internet Control Message Protocol version 4
     /// packet.
-    pub fn emit<T: AsRef<[u8]> + AsMut<[u8]> + ?Sized>(&self, packet: &mut Packet<&mut T>) {
+    pub fn emit<T: AsRef<[u8]> + AsMut<[u8]> + ?Sized>(&self, 
+                                                       packet: &mut Packet<&mut T>, 
+                                                       emit_checksum: Checksum) {
```

Comment:
I think it's worth passing a `&ChecksumCapabilities` around because all of the parse/emit methods already know what kind of packet are they working with, not individual `Checksum` fields, since then the resulting code is easier to read. `header.emit(&mut ip_packet, checksum_caps)` plus `if checksum_caps.ipv4.tx() { ... }` vs `header.emit(&mut ip_packet, emit_checksum)` plus `emit_checksum.tx()`, where `emit_checksum` actually conatins both emission and parsing capabilities in the latter case.